Source/Urho3D/Core/Variant.h

```diff
@@ -1364,23 +1371,22 @@ class URHO3D_API Variant
     VariantMap* GetVariantMapPtr() { return type_ == VAR_VARIANTMAP ? value_.variantMap_ : nullptr; }
 
     /// Return a pointer to a modifiable custom variant value or null on type mismatch.
-    template <class T> T* GetCustomPtr()
-    {
-        if (CustomVariantValue* value = GetCustomVariantValuePtr())
-        {
-            if (value->IsType<T>())
-                return value->GetValuePtr<T>();
-        }
-        return nullptr;
-    }
+    template <class T> T* GetCustomPtr() { return const_cast<T*>(const_cast<const Variant*>(this)->GetCustomPtr<T>()); }
```

Comment:
This smells. Cant non-const and const GetCustomPtr() bodies be swapped? So it does all the work in non-const context and non-const result can always be returned from const method.